### PR TITLE
Direct audio

### PR DESCRIPTION
**Pull request by eeejay**
This branch changes the spec and client library to enable the client to stream the audio to the device directly. Instead of providers writing to the audio output device they write to a pipe. The client reads from the pipe and does the audio output itself.

### CODE REVIEW

**Comment by eeejay** [comment]
@ford-prefect, this isn't a small change I don't think I need you to look at everything. I think the important bits are the gstreamer pipeline in `SpielSpeaker` ([300b93e](https://github.com/eeejay/spiel/pull/23/commits/300b93e518618690296fae08dc0157c0d6a81b73)), the new `SpielProvider` library ([4e35875](https://github.com/eeejay/spiel/pull/23/commits/4e358755820b6e89d157c67e7eb0da4da0d963e1)), and the gstreamer source element I created for receiving the audio/event chunks in the client ([5b06f4f](https://github.com/eeejay/spiel/pull/23/commits/5b06f4fda276403a31abc13a85ae7af85077d7eb) and [2f68d7b](https://github.com/eeejay/spiel/pull/23/commits/2f68d7b10b6d7b992561305a13b1f18733c4d6c9)).

Thank you for your input!

**Comment by eeejay** [comment]
I'm going to go ahead and merge this. Hopefully someone who knows a thing or two about streams ans audio can look at this project! This fixes issue #8.

File: libspiel/spiel-provider-src.c [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  SpielProviderSrc *src = SPIEL_PROVIDER_SRC (bsrc);
+  gboolean got_header =
+      spiel_provider_stream_reader_get_stream_header (src->reader);
+  return got_header;
+}
+
+static gboolean
+spiel_provider_src_stop (GstBaseSrc *bsrc)
+{
+  // XXX: Do we need this?
```
Comment (ford-prefect) [inline libspiel/spiel-provider-src.c:134]: Pretty much all of these are optional, so you can skip these if they're noops.

**Review by ford-prefect — COMMENTED** [review]
I took a quick look at the source and speaker code, all good. I'll try to take a look at the stream reader/writer, but that might only happen a bit later.

**Pull request merged**